src/store/RelayRecordStore.js

```diff
@@ -124,11 +126,12 @@ class RelayRecordStore {
         'cannot be null or undefined.',
         storageKey
       );
-      return identifyingArgValue;
+      return stringifyArg(identifyingArgValue);
     }
     if (identifyingArgValue == null) {
       identifyingArgValue = EMPTY;
     }
+    identifyingArgValue=stringifyArg(identifyingArgValue);
```

Comment:
Since this is basically a hash of an identifying argument and not the value itself, consider:

```js
const identifyingArgHash = stableStringify(identifyingArgValue);
```